### PR TITLE
clear errors when apply new filters

### LINKED ISSUE #8538: No ability to reset filters after an error
If a filter is applied to the Product Categories window, if that filter is removed, it generates a "Something went wrong" error, that prevents you from ever "erasing" the problematic error.

### Preconditions
1. Magento 2.4-develop
2. PHP 7.3
3. Nginx hosted service

### Steps to reproduce
1. Create a new store view (call it test for example)
2. Log into admin > Products > Catalog 
3. Set up and apply a filter for the newly created store view;
4. Now, go to Admin->Stores->All Stores;
5. Delete the created before "test" store view;
6. Switch back to the Product Catalog, you'll get "Something went wrong" error

### Expected result
1. If there is an error in selecting a filter option, it should apply NO FILTER at all, and indicate that a filter option was broken.

### Actual result

:heavy_multiplication_x: _Attention. Something went wrong_
Screenshot from 2020-04-08 11-34-07

### PR DESCRIPTION
**Pull request by AbdulRahmanAbouzaid**
### Description (*)

Clear error when applying new filters instead of still displaying it **which is the wrong behavior** as mentioned in this [comment](https://github.com/magento/magento2/issues/8538#issuecomment-610846049)
### Fixed Issues (if relevant)

1. Fixes magento/magento2#8538

### Manual testing scenarios (*)

1. Create a new store view (call it "Second Store View" for example);
2. Log in to Admin -> Catalog -> Products;
3. Set up and apply a filter for the newly created store view;
image
4. Now, go to Admin->Stores->All Stores;
5. Delete the created before "Second Store View" store view;
6. Go to Admin->Catalog->Products again;
7.  Set up and apply a new filter and the error message will be disappeared.

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by AbdulRahmanAbouzaid** [comment]
@magento give me test instance

**Comment by magento-engcom-team** [comment]
Hi @AbdulRahmanAbouzaid. Thank you for your request. I'm working on Magento instance for you

**Comment by AbdulRahmanAbouzaid** [comment]
@magento run all test

**Comment by AbdulRahmanAbouzaid** [comment]
@magento run all tests

**Review by lenaorobei — CHANGES_REQUESTED** [review]
Thanks for this fix. Please add MFTF test to cover expected behavior.

**Comment by AbdulRahmanAbouzaid** [comment]
@lenaorobei I've provided the mftf using [AdminGridFilterDeleteAndVerifyErrorMessageTest](https://github.com/magento/magento2/blob/2.4-develop/app/code/Magento/Ui/Test/Mftf/Test/AdminGridFilterDeleteAndVerifyErrorMessageTest.xml) 
Can you review it please?

File: app/code/Magento/Ui/Test/Mftf/Test/AdminGridFilterRemoveErrorMessageBeforeApplyFilters.xml [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+  */
+-->
+<!-- Test XML Example -->
+<tests xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="urn:magento:mftf:Test/etc/testSchema.xsd">
+    <test name="AdminGridFilterRemoveErrorMessageBeforeApplyFilters">
+        <annotations>
+            <stories value="Reset Error Messages"/>
+            <title value="Remove Error Message Before Apply Filters"/>
+            <description value="Test log in to uI and Remove Error Message Before Apply Filters"/>
+            <testCaseId value="MC-142721"/>
```
Comment (lenaorobei) [inline app/code/Magento/Ui/Test/Mftf/Test/AdminGridFilterRemoveErrorMessageBeforeApplyFilters.xml]:
```suggestion

```

File: app/code/Magento/Ui/Test/Mftf/Test/AdminGridFilterRemoveErrorMessageBeforeApplyFilters.xml [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+  * See COPYING.txt for license details.
+  */
+-->
+<!-- Test XML Example -->
+<tests xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="urn:magento:mftf:Test/etc/testSchema.xsd">
+    <test name="AdminGridFilterRemoveErrorMessageBeforeApplyFilters">
+        <annotations>
+            <stories value="Reset Error Messages"/>
+            <title value="Remove Error Message Before Apply Filters"/>
+            <description value="Test log in to uI and Remove Error Message Before Apply Filters"/>
```
Comment (lenaorobei) [inline app/code/Magento/Ui/Test/Mftf/Test/AdminGridFilterRemoveErrorMessageBeforeApplyFilters.xml]:
```suggestion
            <description value="Test Login to Admin UI and Remove Error Message Before Apply Filters"/>
```

File: app/code/Magento/Ui/Test/Mftf/Test/AdminGridFilterRemoveErrorMessageBeforeApplyFilters.xml [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+            <testCaseId value="MC-142721"/>
+            <severity value="CRITICAL"/>
+            <group value="uI"/>
```
Comment (lenaorobei) [inline app/code/Magento/Ui/Test/Mftf/Test/AdminGridFilterRemoveErrorMessageBeforeApplyFilters.xml]:
```suggestion
            <group value="ui"/>
```

File: app/code/Magento/Ui/Test/Mftf/Test/AdminGridFilterRemoveErrorMessageBeforeApplyFilters.xml [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+            <group value="mtf_migrated"/>
```
Comment (lenaorobei) [inline app/code/Magento/Ui/Test/Mftf/Test/AdminGridFilterRemoveErrorMessageBeforeApplyFilters.xml]:
```suggestion
```

File: app/code/Magento/Ui/Test/Mftf/Test/AdminGridFilterRemoveErrorMessageBeforeApplyFilters.xml [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+        </annotations>
+
+        <before>
+            <magentoCLI command="config:set system/backup/functionality_enabled 1" stepKey="setEnableBackupToYes"/>
```
Comment (lenaorobei) [inline app/code/Magento/Ui/Test/Mftf/Test/AdminGridFilterRemoveErrorMessageBeforeApplyFilters.xml]: I believe this is redundant.

File: app/code/Magento/Ui/Test/Mftf/Test/AdminGridFilterRemoveErrorMessageBeforeApplyFilters.xml [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+                <argument name="rootCategory" value="$$rootCategory.name$$"/>
+            </actionGroup>
+            <!-- Create second store view -->
+            <actionGroup ref="AdminCreateStoreViewActionGroup" stepKey="createCustomStoreView">
+                <argument name="StoreGroup" value="customStoreGroup"/>
+                <argument name="customStore" value="customStoreEN"/>
+            </actionGroup>
+        </before>
+        <after>
+            <magentoCLI command="config:set system/backup/functionality_enabled 0" stepKey="setEnableBackupToNo"/>
```
Comment (lenaorobei) [inline app/code/Magento/Ui/Test/Mftf/Test/AdminGridFilterRemoveErrorMessageBeforeApplyFilters.xml]: Same here.

**Review by lenaorobei — CHANGES_REQUESTED** [review]
Thanks a lot!

Please see minor review comments and suggestions.

**Comment by AbdulRahmanAbouzaid** [comment]
@magento run all tests

**Comment by lenaorobei** [comment]
@magento run Functional Tests EE

**Review by lenaorobei — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @lenaorobei, thank you for the review.
[ENGCOM-8098](https://jira.corp.magento.com/browse/ENGCOM-8098) has been created to process this Pull Request
:eight_spoked_asterisk: @lenaorobei, could you please add one of the following labels to the Pull Request?

| Label | Description |
|--------------------------------|-----------------------------------------------------------------|
| ```Auto-Tests: Covered``` | All changes in Pull Request is covered by auto-tests |
| ```Auto-Tests: Not Covered``` | Changes in Pull Request requires coverage by auto-tests |
| ```Auto-Tests: Not Required``` | Changes in Pull Request does not require coverage by auto-tests |

**Comment by engcom-Oscar** [comment]
:heavy_check_mark: QA Passed

Tested according manual scenario described in ticket 

### Before :heavy_multiplication_x: 
Error message persists after a new filter is submitted


### After :heavy_check_mark:
Error message disappeared.

**Comment by engcom-Charlie** [comment]
@magento run all tests

**Pull request merged**